bench_method.go

```diff
@@ -65,34 +70,39 @@ func (m benchmarkMethod) call(t transport.Transport) (time.Duration, error) {
 	return duration, err
 }
 
-func peerBalancer(peers []string) func(i int) string {
+func peerBalancer(peers []string) func(i int) (string, int) {
 	numPeers := len(peers)
 	startOffset := rand.Intn(numPeers)
-	return func(i int) string {
+	return func(i int) (string, int) {
 		offset := (startOffset + i) % numPeers
-		return peers[offset]
+		return peers[offset], offset
 	}
 }
 
 // WarmTransports returns n transports that have been warmed up.
 // No requests may fail during the warmup period.
-func (m benchmarkMethod) WarmTransports(n int, tOpts TransportOptions, warmupRequests int) ([]transport.Transport, error) {
+func (m benchmarkMethod) WarmTransports(n int, tOpts TransportOptions, warmupRequests int) ([]peerTransport, error) {
 	tOpts, err := loadTransportPeers(tOpts)
 	if err != nil {
 		return nil, err
 	}
 
 	peerFor := peerBalancer(tOpts.Peers)
-	transports := make([]transport.Transport, n)
+	transports := make([]peerTransport, n)
 	errs := make([]error, n)
 
 	var wg sync.WaitGroup
 	for i := range transports {
 		wg.Add(1)
 		go func(i int, tOpts TransportOptions) {
 			defer wg.Done()
-			tOpts.Peers = []string{peerFor(i)}
-			transports[i], errs[i] = m.WarmTransport(tOpts, warmupRequests)
+
+			peerHostPort, peerIndex := peerFor(i)
+			tOpts.Peers = []string{peerHostPort}
+
+			tp, err := m.WarmTransport(tOpts, warmupRequests)
+			transports[i] = peerTransport{tp, peerIndex}
```

Comment:
Do linters require this for private types within the same package? Named fields are necessary when it's an external type (since adding a new field will break otherwise), and when dealing with multiple fields with similar types. In this case, since we have unique types, and it's a type declared within the package (and it's unexported), is there much value in field names?

Reply:
I guess not if the variable names are descriptive enough (which they are in this case).